dask/dataframe/io/parquet/arrow.py

```diff
@@ -952,18 +934,10 @@ def _collect_dataset_info(
         # Get all partition keys (without filters) to populate partition_obj
         partition_obj = []  # See `partition_info` description below
         hive_categories = defaultdict(list)
-        file_frag = None
-        for file_frag in ds.get_fragments():
-            if partitioning_supported:
-                # Can avoid manual category discovery for pyarrow>=5.0.0
-                break
-            keys = pa_ds._get_partition_keys(file_frag.partition_expression)
-            if not (keys or hive_categories):
-                break  # Bail - This is not a hive-partitioned dataset
-            for k, v in keys.items():
-                if v not in hive_categories[k]:
-                    hive_categories[k].append(v)
-
+        try:
+            file_frag = next(ds.get_fragments())
+        except StopIteration:
+            file_frag = None
```

Comment:
@rjzamora this change should be equivalent to what we currently have. Though there are

```
# TODO: Remove this after pyarrow>=5.0.0 is required
```

and
```
# Use ds.partitioning for pyarrow>=5.0.0
```

comments a few lines below that indicates we should be able to remove more compatibility code. I'm not very familiar with this specific section of the arrow engine, so I've left those sections in for the time being (I don't want to break anything). Any interest in ripping out more code? Either directly in this PR, or in a follow-up, whichever you prefer

Reply:
I was looking at the same code while working through https://github.com/dask/dask/pull/10007 and https://github.com/dask/dask/issues/9968 - Pretty sure this change is safe, but I'd be happy to take responsibility for cleaning out stale code in a separate PR if it's helpful

Reply:
I suppose there are quite a few parquet PRs in flight (https://github.com/dask/dask/pull/9885 is a particularly high-priority one for RAPIDS), so I suppose I need to start chipping away at the backlog a bit more aggressively :)

Reply:
Awesome, thanks @rjzamora! 

Is there a preferred merge order with this PR and https://github.com/dask/dask/pull/9885 to minimize merge conflicts?